datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/authorization/AuthorizationUtils.java

```diff
@@ -33,6 +61,32 @@ public static boolean canManageGlossaries(@Nonnull QueryContext context) {
     return isAuthorized(context, Optional.empty(), PoliciesConfig.MANAGE_GLOSSARIES_PRIVILEGE);
   }
 
+  /**
+   * Returns true if the current used is able to create Tags. This is true if the user has the 'Manage Tags' or 'Create Tags' platform privilege.
+   */
+  public static boolean canCreateTags(@Nonnull QueryContext context) {
+    final DisjunctivePrivilegeGroup orPrivilegeGroups = new DisjunctivePrivilegeGroup(
+        ImmutableList.of(
+            new ConjunctivePrivilegeGroup(ImmutableList.of(
+            PoliciesConfig.CREATE_TAGS_PRIVILEGE.getType())),
+            new ConjunctivePrivilegeGroup(ImmutableList.of(
+                PoliciesConfig.MANAGE_TAGS_PRIVILEGE.getType()))
+        ));
+
+    return AuthorizationUtils.isAuthorized(
+        context.getAuthorizer(),
+        context.getActorUrn(),
+        orPrivilegeGroups);
+  }
+
+  public static boolean canManageTags(@Nonnull QueryContext context) {
+    return isAuthorized(context, Optional.empty(), PoliciesConfig.MANAGE_TAGS_PRIVILEGE);
+  }
+
+  public static boolean canDeleteEntity(@Nonnull Urn entityUrn, @Nonnull QueryContext context) {
+    return isAuthorized(context, Optional.of(new ResourceSpec(entityUrn.getEntityType(), entityUrn.toString())), PoliciesConfig.DELETE_ENTITY_PRIVILEGE);
```

Comment:
so the user has to have this `DELETE_ENTITY_PRIVILEGE` specifically for the entity type that you pass in right? aka it's possible to have this privilege for Tags but not Domains

Reply:
yes correct!